---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures"
)
```

# R package `AEenrich`

`r badger::badge_devel("umich-biostatistics/AEenrich", "blue")` 
`r badger::badge_code_size("umich-biostatistics/AEenrich")` 
`r badger::badge_cran_download("AEenrich", "grand-total", "blue")` 
`r badger::badge_cran_download("AEenrich", "last-month", "green")`
`r badger::badge_custom("Read on", "arXiv", "orange", "https://arxiv.org/abs/2007.02266")`
`r badger::badge_cran_checks("AEenrich")`

## Overview

Perform Adverse Event Enrichment Tests. The enrich function is used to perform Adverse event (AE) enrichment analysis. Unlike the continuous gene expression data, AE data are counts. Therefore, AE data has many zeros and ties. We propose two enrichment tests. AEFisher is a modified Fisher's exact test based on pre-selected significant AEs, while AEKS is based on a modified Kolmogorov-Smirnov statistic.

## Install from CRAN

```{r, eval = FALSE}
install.packages("AEenrich")
```

Then, load the package with

```{r, eval = FALSE}
library(AEenrich)
```

## Install from Github

If the devtools package is not yet installed, install it first:

```{r, eval=FALSE}
install.packages('devtools')
```

Then run: 

```{r, eval = FALSE}
# install AEenrich from Github:
devtools::install_github('umich-biostatistics/AEenrich') 
library(AEenrich)
```

## Example usage

For documentation pages:

```{r, eval = FALSE}
?AEenrich
?enrich
?count_cases
```

### Quick example: 

```{r, eval = FALSE}
# AEKS
## Type I data: data on report level
enrich(data = covid1, covar = c("SEX", "AGE"), p = 0, method = "aeks",
       n_perms = 1000, drug.case = "COVID19", dd.group = group_info,
       drug.control = "OTHER", min_size = 5, min_AE = 10, zero = FALSE)
## Type II data: aggregated data
enrich(data = covid2, covar = c("SEX", "AGE"), p = 0, method = "aeks",
       n_perms = 1000, drug.case = "DrugYes", dd.group = group_info,
       drug.control = "DrugNo", min_size = 5, min_AE = 10)
# AEFISHER
## Type I data: data on report level
enrich(data = covid1, covar = c("SEX", "AGE"), p = 0, method = "aefisher",
       n_perms = 1000, drug.case = "COVID19", dd.group = group_info,
       drug.control = "OTHER", min_size = 5, min_AE = 10, q.cut = 0.05, 
       or.cut = 1.5, cores = 8)
## Type II data: aggregated data
enrich(data = covid2, covar = c("SEX", "AGE"), p = 0, method = "aefisher",
       n_perms = 1000, drug.case = "DrugYes", dd.group = group_info,
       drug.control = "DrugNo", min_size = 5, min_AE = 10)

## Convert type I data to type II data
count_cases (data = covid1, drug.case = "COVID19", drug.control = "OTHER",
             covar_cont = c("AGE"), covar_disc = c("SEX"),
             breaks = list(c(16,30,50,65,120)))

```

### Current Suggested Citation

1. Li, S. and Zhao, L. (2020). Adverse event enrichment tests using VAERS. [arXiv:2007.02266](https://arxiv.org/abs/2007.02266).

2. Subramanian, A.e.a. (2005). Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A. Proceedings of the National Academy of Sciences. 102. 15545-15550. 

3. Tian, Lu & Greenberg, Steven & Kong, Sek Won & Altschuler, Josiah & Kohane, Isaac & Park, Peter. (2005). Discovering statistically significant pathways in expression profiling studies. Proceedings of the National Academy of Sciences of the United States of America. 102. 13544-9. 10.1073/pnas.0506577102. 